### PR TITLE
Decentralizing: Removing big centralized Type

### PR DESCRIPTION
**Pull request by Yoric**
6th big Decentralization PR. This one gets us rid of enum `Type` and replaces it with a trait `Format`.
I hope that the 7th will be the final PR.

### CODE REVIEW

**Comment by Yoric** [comment]
r? @cr for Philips Hue.
r? @dhylands for IP Camera.
r? @julienw for OpenZWave
r? @mcav for Thinkerbell
r? @fabricedesre for Taxonomy Router
r? @aosmond for WebPush

No deep changes in client code, nor REST API changes this time.

**Comment by fabricedesre** [comment]
r=me for the taxonomy_router
thanks!

**Comment by aosmond** [comment]
r=me for webpush. the 7th PR will fix up the REST API right?

**Comment by Yoric** [comment]
> r=me for webpush. the 7th PR will fix up the REST API right?

Actually, no. 7th PR is for decentralizing `Value`. For the REST API, I don't have good ideas yet, so any input is welcome.

**Comment by mcav** [comment]
lgtm for thinkerbell

File: src/adapters/thinkerbell/mod.rs [hunk 1]
```diff
@@ -412,3 +407,16 @@ impl ThinkerbellAdapter {
...
+struct ThinkerbellRuleFormat;
+impl io::Format for ThinkerbellRuleFormat {
+    fn description(&self) -> String {
+        "Thinkerbell Rule".to_owned()
+    }
+}
+
+lazy_static! {
+    pub static ref THINKERBELL_RULE: Arc<io::Format> = Arc::new(ThinkerbellRuleFormat);
+}
```
Comment (cr) [inline src/adapters/thinkerbell/mod.rs:422]: Nit: no newline

**Comment by cr** [comment]
r+ for Hue changes.

**Comment by julienw** [comment]
r=me, but be careful that there seems to be are build issues on some platforms

**Comment by Yoric** [comment]
Ah, right, minor snafu on TTS (which doesn't get built on Mac). I'll fix this asap.

**Pull request merged**